### PR TITLE
Implement to/from dict helpers for PropertyInfo/MethodInfo

### PR DESCRIPTION
**Pull request by Naros**
Others may appreciate these rather than create their own implementations.

### CODE REVIEW

File: include/godot_cpp/core/property_info.hpp [hunk 1]
```diff
@@ -72,6 +72,40 @@ struct PropertyInfo {
...
+	}
+
+	static PropertyInfo from_dict(const Dictionary &p_dict) {
+		PropertyInfo pi;
+		if (p_dict.has("name")) {
+			pi.name = p_dict["name"];
+		}
+		if (p_dict.has("name")) {
+			pi.name = p_dict["name"];
+		}
```
Comment (AThousandShips) [inline include/godot_cpp/core/property_info.hpp:93]: Repeated

Comment (Naros) [reply include/godot_cpp/core/property_info.hpp:93]: Thanks, no idea how that happened, cut-n-paste gremlin 👀 .  Fixed on the forced push 👍 .

**Review by dsnopek — APPROVED** [review]
Thanks! Looks good to me :-)

This code appears to match the implementations of the same methods in Godot itself.

**Comment by dsnopek** [comment]
Cherry-picked for 4.2 in PR https://github.com/godotengine/godot-cpp/pull/1465

**Comment by dsnopek** [comment]
Cherry-picked for 4.1 in PR https://github.com/godotengine/godot-cpp/pull/1466

**Comment by hayahane** [comment]
Found an unexpected [bug](https://github.com/limbonaut/limboai/pull/364) caused by the override Dictionary operator.
image
Is there a way to still have these handy helpers and make them explicit? Do we do full conversion most of the time?

**Pull request merged**